diff -rpux webgui.log ./lib/WebGUI/Content/AssetManager.pm ../webgui-2010-06-14/lib/WebGUI/Content/AssetManager.pm
--- ./lib/WebGUI/Content/AssetManager.pm	2010-06-15 08:22:05.628542554 +0400
+++ ../webgui-2010-06-14/lib/WebGUI/Content/AssetManager.pm	2010-06-14 16:49:30.325562737 +0400
@@ -8,7 +8,6 @@ use WebGUI::Form;
 use WebGUI::Paginator;
 use WebGUI::Utility;
 use WebGUI::Macro::AdminBar;
-use WebGUI::Form::CsrfToken;
 
 #----------------------------------------------------------------------------
 
Only in ./lib/WebGUI/Session: .Form.pm.swp
Only in ../webgui-2010-06-14/lib/WebGUI/Session: .Http.pm.swp
diff -rpux webgui.log ./lib/WebGUI/Workflow/Activity/ExpirePurchasedThingyRecords.pm ../webgui-2010-06-14/lib/WebGUI/Workflow/Activity/ExpirePurchasedThingyRecords.pm
--- ./lib/WebGUI/Workflow/Activity/ExpirePurchasedThingyRecords.pm	2010-06-14 19:03:22.120405098 +0400
+++ ../webgui-2010-06-14/lib/WebGUI/Workflow/Activity/ExpirePurchasedThingyRecords.pm	2010-06-06 05:33:25.431762140 +0400
@@ -16,7 +16,6 @@ package WebGUI::Workflow::Activity::Expi
 
 use strict;
 use base 'WebGUI::Workflow::Activity';
-use WebGUI::AssetCollateral::Sku::ThingyRecord::Record;
 
 =head1 NAME
 
diff --git a/lib/WebGUI/Session/Http.pm b/lib/WebGUI/Session/Http.pm
index 220d1da..544539f 100644
--- a/lib/WebGUI/Session/Http.pm
+++ b/lib/WebGUI/Session/Http.pm
@@ -327,6 +327,7 @@ sub sendHeader {
 		}
 		$request->status($self->getStatus());
 		$request->status_line($self->getStatus().' '.$self->getStatusDescription());
+    $request->send_http_header;
 	}
 	return undef;
 }
