--- Fake.pm	2002-02-01 18:00:52.000000000 +0300
+++ /usr/local/jail/internal/usr/local/lib/perl5/site_perl/5.8.9/Apache/Fake.pm	2010-09-06 23:41:31.680037065 +0400
@@ -1076,10 +1076,6 @@ sub new
     # http protocol
     $conf{'PROTOCOL'} = $conf{'ENV'}{'SERVER_PROTOCOL'};
 
-    # path info
-    $conf{'PATH_INFO'} = substr($conf{'ENV'}{'PATH_TRANSLATED'},length($conf{'FILE'}));
-    $conf{'ENV'}{'PATH_INFO'} = $conf{'PATH_INFO'};
-
     # time
     $conf{'TIME'} = time();
 
@@ -1126,8 +1122,6 @@ sub new
 
     # connection
     $conf{'ABORTED'} = 0;
-    $conf{'FD_IN'} = fileno(STDIN);
-    $conf{'FD_OUT'} = fileno(STDOUT);
 
     # remote ident
     $conf{'REMOTE_IDENT'} = $conf{'ENV'}{'REMOTE_IDENT'};
@@ -1265,27 +1259,13 @@ sub new
     	die ($@) if $@;
     }
 
-    die 'no PerlHandler found, but we have: '.join(keys %$handlers) unless exists $$handlers{'PerlHandler'};
-
-    my $eval_string = $$handlers{'PerlHandler'};
-    if ($eval_string =~ m/->/) {
-	$eval_string .= '($r)';
-    } elsif ($eval_string =~ m/^[a-zA-Z_0-9:]+$/) {
-	$eval_string .= '::handler($r)';
-    } elsif ($eval_string !~ m/[{&]/) {
-	die "unknown handler syntax: $eval_string";
-    }
 
     %ENV = %{$conf{'ENV'}};
 
     #$r->warn("invoking: $eval_string");
-    my $rc = eval($eval_string);
     die ($@) if $@;
-    #$r->warn("rc = $rc");
-    $r->status($rc) if $rc;
-    $r->send_http_header;
 
-    return $r;
+      return $r;
 }
 
 1;
